### PR TITLE
Add GraphQL mutation to enroll identities

### PR DESCRIPTION
**Pull request by sduenas**
This mutation allows to enroll identities in organizations. Some functions were added to the API to support this feature.

This commit adds the features requested on #185

### CODE REVIEW

File: sortinghat/core/db.py [hunk 1]
```diff
@@ -378,6 +427,71 @@ def to_none_if_empty(x):
...
+    This function returns as result a new `Enrollment` object.
+
+    :param uidentity: unique identity to enroll
+    :param organization: organization where the unique identity is enrolled
+    :param start: date when the enrollment starts
+    :param end: date when the enrollment ends
+
+    :returns: a new enrollment
+
+    :raises ValeError: when either `start` or `end` are `None`;
```
Comment (valeriocos) [inline sortinghat/core/db.py]: typo -> ValeError

File: sortinghat/core/api.py [hunk 2]
```diff
@@ -246,3 +253,93 @@ def update_profile(uuid, **kwargs):
...
+    :param to_date: date when the enrollment ends
+
+    :returns: a unique identity with enrollment data updated
+
+    :raises NotFoundError: when either `uuid` or `organization` are not
+        found in the registry.
+    :raises InvalidValeError: raised in three cases, when either identity or
+        organization are None or empty strings; when "from_date" < 1900-01-01 or
+        "to_date" > 2100-01-01; when "from_date > to_date"
+    :raises AlreadyExistsError: raised when given period for that enrollment
```
Comment (valeriocos) [inline sortinghat/core/api.py]: typo -> missing the between when and given ?

File: tests/test_api.py [hunk 3]
```diff
@@ -777,3 +783,314 @@ def test_gender_acc_invalid_range(self):
 
         with self.assertRaisesRegex(InvalidValueError, msg):
             api.update_profile(uuid, gender='male', gender_acc=101)
+
+
+class TestEnroll(TestCase):
+    """Unit tests for enroll"""
+
+    def test_enroll(self):
+        """Check whether it adds an enrollments to a unique identity and a organization"""
```
Comment (valeriocos) [inline tests/test_api.py]: typo -> an organization

File: tests/test_utils.py [hunk 4]
```diff
@@ -19,14 +19,333 @@
...
 
+CANT_COMPARE_DATES_ERROR = "can't compare offset-naive and offset-aware datetimes"
+DATE_OUT_OF_BOUNDS_ERROR = "'{type}' date {date} is out of bounds"
 UNACCENT_TYPE_ERROR = "argument must be a string; int given"
 
 
+class TestMergeDatetimeRanges(TestCase):
+    """Unit tests for merge_datetime_ranges function"""
+
+    def test_merge_datetime_ranges(self):
```
Comment (valeriocos) [inline tests/test_utils.py]: The test method `test_merge_datetime_ranges` contains many cases, maybe they could be splitted in different test methods. Another option could be to add some details in the line comments (`# Case 1: ...., `# Case 2: ..., etc.)

File: tests/test_utils.py [hunk 5]
```diff
@@ -19,14 +19,333 @@
...
+            (datetime.datetime(1900, 1, 1, tzinfo=UTC), datetime.datetime(2100, 1, 1, tzinfo=UTC))
+        ]
+
+        ranges = [r for r in merge_datetime_ranges(dates)]
+        expected = [
+            (datetime.datetime(1900, 1, 1, tzinfo=UTC), datetime.datetime(2100, 1, 1, tzinfo=UTC))
+        ]
+        self.assertListEqual(ranges, expected)
+
+    def test_merge_excluding_limits(self):
```
Comment (valeriocos) [inline tests/test_utils.py]: Same comments as above

File: sortinghat/core/db.py (hunk 1 above)
Comment (sduenas) [reply sortinghat/core/db.py]: It should be fixed with the latest changes.

File: sortinghat/core/api.py (hunk 2 above)
Comment (sduenas) [reply sortinghat/core/api.py]: It should be fixed with the latest changes.

File: tests/test_api.py (hunk 3 above)
Comment (sduenas) [reply tests/test_api.py]: It should be fixed with the latest changes.

File: tests/test_utils.py (hunk 4 above)
Comment (sduenas) [reply tests/test_utils.py]: I mainly followed the approach of commenting each case. It was hard to describe each case. In some cases it was more clear I create new test cases.

File: tests/test_utils.py (hunk 5 above)
Comment (sduenas) [reply tests/test_utils.py]: Same as above.

**Comment by sduenas** [comment]
@valeriocos please check the new version of this PR when you can.

File: tests/test_utils.py (hunk 4 above)
Comment (valeriocos) [reply tests/test_utils.py]: perfect!

**Comment by valeriocos** [comment]
thank you for addressing the comments @sduenas , the PR LGTM

**Pull request merged**